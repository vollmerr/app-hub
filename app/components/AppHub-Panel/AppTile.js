import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Icon } from 'office-ui-fabric-react/lib/Icon';

import Link from 'components/Link';
import theme from 'utils/theme';

const App = styled(Link) `
  display: flex;
  float: left;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin: 3px;
  height: 102px;
  width: 102px;
  box-sizing: border-box;
  overflow: visible;
  transition: .25s ease;
  pointer-events: all;
  font-size: ${(props) => props.size || '18px'};
  text-decoration: none;
  background: ${theme.themeDarkAlt};
  box-shadow: 0 0 2px ${theme.themeDarkAlt};

  &:hover {
    background: ${theme.white};
    box-shadow: 0 0 5px ${theme.white};
  }

  * {
    color: white;
    display: flex;
  }

  &:hover * {
    color: ${theme.themePrimary};
  }
`;

const StyledIcon = styled(Icon) `
  font-size: 26px;
`;

const Text = styled.div`
  display: table;
`;

class AppTile extends React.PureComponent {
  render() {
    const { onClick, route } = this.props;
    const { path, href, name, icon } = route;

    return (
      <App to={path} href={href} onClick={onClick}>
        <StyledIcon iconName={icon} />
        <Text>{name}</Text>
      </App>
    );
  }
}

AppTile.propTypes = {
  onClick: PropTypes.func,
  route: PropTypes.object.isRequired,
};

export default AppTile;
