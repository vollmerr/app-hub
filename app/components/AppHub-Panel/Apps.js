/**
*
* Apps
*
*/

import React from 'react';

function Apps() {
  return (
    <div>
      IN Apps PANEL
    </div>
  );
}

Apps.propTypes = {

};

export default Apps;
