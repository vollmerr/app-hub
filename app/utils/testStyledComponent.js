import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

function testStyledComponent(Component, extend = null) {
  describe(`<${Component.displayName} />`, () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Component />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it(`should render an <${Component.target}> tag`, () => {
      expect(wrapper.type()).toEqual(Component.target);
    });

    it('should have a className attribute', () => {
      expect(wrapper.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      wrapper.setProps({ id });
      expect(wrapper.prop('id')).toEqual(id);
    });

    if (extend) {
      it(`should extend ${extend.name}`, () => {
        expect(wrapper.find(extend).exists()).toEqual(true);
      });
    } else {
      it('should not adopt an invalid attribute', () => {
        const attribute = 'test';
        wrapper.setProps({ attribute });
        expect(wrapper.prop('attribute')).toBeUndefined();
      });
    }
  });
}

export default testStyledComponent;

