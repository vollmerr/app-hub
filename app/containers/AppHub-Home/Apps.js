/**
*
* Apps
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import AppTile from 'components/AppHub-Panel/AppTile';
import theme from 'utils/theme';

export const Wrapper = styled.div`
  max-width: calc(${theme.hub.numApps} * ${theme.hub.tileSize} + ${(theme.hub.numApps) * 2} * 3px);
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  height: 102px;
  overflow: hidden;
  width: 100%;
  margin-top: 30px;
`;


class Apps extends React.PureComponent {
  render() {
    return (
      <Wrapper>
        {
          this.props.routes.slice(0, theme.hub.numApps).map((route) => (
            <AppTile key={route.key} route={route} />
          ))
        }
      </Wrapper>
    );
  }
}

Apps.propTypes = {
  routes: PropTypes.array.isRequired,
};

export default Apps;
