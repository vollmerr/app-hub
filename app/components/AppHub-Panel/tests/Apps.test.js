import React from 'react';
import { shallow } from 'enzyme';

import testStyledComponent from 'utils/testStyledComponent';

import AppTile from '../AppTile';
import Apps, { Wrapper } from '../Apps';

testStyledComponent(Wrapper);

const onClick = jest.fn();
const routes = [
  { name: 'route 1', key: 'test key 1' },
  { name: 'route 2', key: 'test key 2' },
];

describe('<Apps />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Apps onClick={onClick} routes={routes} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render a h2 header', () => {
    expect(wrapper.find('h2').length).toEqual(0);
  });

  it('should render the routes as AppTile`s', () => {
    expect(wrapper.find(AppTile).length).toEqual(routes.length);
  });
});
