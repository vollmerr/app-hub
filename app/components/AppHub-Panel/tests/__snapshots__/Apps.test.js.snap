// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Apps /> should render correctly 1`] = `
<Apps__Wrapper>
  <AppTile
    key="test key 1"
    onClick={[Function]}
    route={
      Object {
        "key": "test key 1",
        "name": "route 1",
      }
    }
  />
  <AppTile
    key="test key 2"
    onClick={[Function]}
    route={
      Object {
        "key": "test key 2",
        "name": "route 2",
      }
    }
  />
</Apps__Wrapper>
`;

exports[`<Apps__Wrapper /> should render correctly 1`] = `
.c0 {
  padding: 15px;
  background: #004578;
  height: 100%;
}

<div
  className="c0"
/>
`;
